src/internal/log/http.go

```diff
@@ -47,10 +47,10 @@ func AddLoggerToHTTPServer(rctx context.Context, name string, s *http.Server) {
 					zap.String("peer", r.RemoteAddr),
 					id,
 				}
-				Info(ctx, "incoming http request", fields...)
+				Info(reqCtx, "incoming http request", fields...)
 			}
 
-			ctx = ChildLogger(ctx, "", WithFields(id))
+			ctx = ChildLogger(reqCtx, "", WithFields(id))
 			ctx = metadata.NewOutgoingContext(ctx, metadata.MD{"x-request-id": requestID})
```

Comment:
This is now deleting the logger context.

I try to always name the most relevant context `ctx` to avoid this refactoring pitfall (that's why some functions take a context.Context named `rctx` ("root context") when the first line of the function does context.WithCancel or something like that.  That way, you don't accidentally pass the wrong context to downstream code.